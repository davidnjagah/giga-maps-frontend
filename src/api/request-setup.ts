import { API_BASE_URL } from '~/env';
import { createRequest } from "~/lib/request";

export const apiBaseUrl = API_BASE_URL;

export const request = createRequest({
  baseUrl: apiBaseUrl,
});