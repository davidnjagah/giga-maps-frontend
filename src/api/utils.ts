import { ServerErrorType } from "~/lib/request/types";

export const errorToasterHandler = (errorResponse: unknown) => {
  const error = errorResponse as ServerErrorType;
  let response = error?.response.data;
  let defaultMessage = 'Something went wrong. Please try again';
  let message: string | undefined;
  if (Array.isArray(response)) {
    response = response[0];
  }
  if (typeof response === 'object') {
    // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
    message = Object.values(response)[0];
    if (Array.isArray(message)) {
      message = message[0];
    }
  } else if (typeof response === 'string') {
    message = response;
  }
  if (typeof message !== 'string') {
    message = defaultMessage;
  }
  return ({
    title: 'Validation error:',
    subtitle: message,
    kind: 'error',
  })
}

export const errorToastFilter = (errorResponse: unknown) => {
  const error = errorResponse as ServerErrorType;
  return error && error.response;
}