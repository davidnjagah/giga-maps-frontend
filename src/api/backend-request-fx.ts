import { createEffect } from 'effector'

import { onCreateNotification } from '~/@/common/Toast/toast.model';
import { onResetLogin } from '~/core/auth/azure-msal/model';
import { ConfigNoFn } from '~/lib/request/types';

import { apiBaseUrl, request } from './request-setup'
import { errorToasterHandler } from './utils';
import { ToastNotificationProps } from '@carbon/react';

export type ApiRequestType = ConfigNoFn

export const backendRequestFx = createEffect(async ({ options, token }: { options: ApiRequestType, token?: string | null; }) => {
  const { headers, baseUrl = `${apiBaseUrl}api/`, ...rest } = options;
  return request({
    baseUrl,
    method: 'GET',
    onResponseError: ({ error }) => {
      const notification = errorToasterHandler(error);
      const response = error.response as { status?: number }
      onCreateNotification(notification as ToastNotificationProps);
      if (response?.status === 401) {
        onResetLogin();
      }
      return Promise.reject(error);
    },
    headers: {
      ...headers,
      ...(token && { Authorization: `Bearer ${token}` }),
    },
    ...rest
  })
});
