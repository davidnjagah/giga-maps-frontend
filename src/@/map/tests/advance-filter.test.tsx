import { fireEvent, render } from "@testing-library/react"

import TextField from "../ui/advanced-filter/text-input"
import FilterPopupContent from "../ui/advanced-filter/filter-popup-content"
import { fetchAdvanceFilterFx } from "~/api/project-connect"
import filterData from "~/tests/data/filter-data"
import RangeTextInput from "../ui/advanced-filter/range-text-input"

describe('AdvancedFilter', () => {

  beforeEach(() => {
    fetchMock.mockResponse((req) => {
      if (req.url.includes('advanced_filters/')) {
        return Promise.resolve(JSON.stringify(filterData))
      } else {
        return Promise.resolve(JSON.stringify(null))
      }
    });
  })
  test('should render Advance filter', async () => {
    await fetchAdvanceFilterFx()
    const { asFragment } = render(<FilterPopupContent setOpen={() => { }} />)
    expect(asFragment()).toMatchSnapshot();
  })


})

describe('TextField with StyledTextInput', () => {
  const mockOnChange = jest.fn();

  test('should call onChange handler when user types in the input', async () => {
    const { getByPlaceholderText } = render(<TextField place_holder="Enter Data Source" itemKey="" value="" parameter={{ label: '', table: '', field: '', filter: '' }} description="" type="" active_countries_list={[]} name="Data Source" onChange={mockOnChange} />);

    const input = getByPlaceholderText('Enter Data Source')

    await fireEvent.change(input, { target: { value: 'New Value' } });

    expect(mockOnChange).toHaveBeenCalledTimes(1);
  });
});

// describe('<SingleDropdown />', () => {
//   const items = [
//     { label: 'All', value: '' },
//     { label: 'Urban', value: 'uAllrban' },
//     { label: 'Rural', value: 'rural' },
//   ];

//   const mockOnChange = jest.fn();

//   it('renders with default "All" selected', () => {
//     const { getByLabelText } = render(
//       <SingleDropdown
//         id="dropdown-education_level"
//         label="All"
//         items={items}
//         onChange={mockOnChange}
//       />
//     );

//     const dropdownLabel = getByLabelText('All');
//     expect(dropdownLabel).toBeInTheDocument();
//     expect(dropdownLabel).toHaveTextContent('All');

//     const selectedItem = getByLabelText('All');
//     expect(selectedItem).toBeInTheDocument();
//     expect(selectedItem).toHaveTextContent('All');
//   });
// });

// describe('MultiSelectDropdown Component', () => {
//   const items = [
//     { label: 'No', value: 'no' },
//     { label: '2G', value: '2g' },
//     { label: '3G', value: '3g' },
//     { label: '4G', value: '4g' },
//   ];

//   const mockProps = {
//     name: 'Coverage Type',
//     parameter: {
//       label: "Coverage Type",
//       table: "school_static",
//       field: "Coverage Type",
//       filter: "in",
//     },
//     choices: items,
//     itemKey: 'coverage_type__in',
//     value: '2g, 3g, 4g',
//     onChange: jest.fn(),
//     type: 'select',
//     description: 'test multiselect dropdown',
//     active_countries_list: null
//   };

//   it('MultiSelectDropdown Component', () => {
//     const { container } = render(<MultiSelectDropdown {...mockProps} />);
//     expect(container).toMatchSnapshot();
//   });

//   it('should handle onChange event', () => {
//     const { getByLabelText } = render(<MultiSelectDropdown {...mockProps} />);
//     const dropdownComponent = getByLabelText(`Select ${mockProps.name}`);

//     fireEvent.change(dropdownComponent, { target: { selectedItems: [{ value: '2g' }] } });
//     expect(mockProps.onChange).toHaveBeenCalledWith('computer_lab__on', '2g');
//   });
// });

describe('RangeTextInput Component', () => {
  const mockProps = {
    name: "No of Teachers",
    type: "range",
    include_none_filter: true,
    active_countries_range: {
      "30": {
        min_place_holder: "Min (0)",
        max_place_holder: "Max (10K)"
      }
    },
    value: {
      none_range: true,
      value: "15,20",
    },
    parameter: {
      label: "No of Teachers",
      table: "school_static",
      field: "num_computers",
      filter: "range",
    },
    itemKey: 'num_computers__range',
    onChange: jest.fn(),
    description: 'test range input',
    active_countries_list: null
  };

  it('RangeTextInput input', () => {
    const { container } = render(<RangeTextInput {...mockProps} />);
    expect(container).toMatchSnapshot();
  });

  it('should handle onChange event for min input (valid value)', () => {
    const { getByPlaceholderText } = render(<RangeTextInput {...mockProps} />);
    const minInput = getByPlaceholderText('Min(0)');

    fireEvent.change(minInput, { target: { value: '25' } });

    expect(mockProps.onChange).toHaveBeenCalled();
  });

  it('should handle onChange event for min input (valid value)', () => {
    const { getByPlaceholderText } = render(<RangeTextInput {...mockProps} value={{ "none_range": false, "value": "" }} />);
    const minInput = getByPlaceholderText('Min(0)');

    fireEvent.change(minInput, { target: { value: '0' } });

    expect(mockProps.onChange).toHaveBeenCalledWith('num_computers__range', { "none_range": false, "value": "" });
  });


  it('should handle onChange event for min input (invalid value)', () => {
    mockProps.onChange.mockReset();
    const { getByPlaceholderText } = render(<RangeTextInput {...mockProps} />);
    const minInput = getByPlaceholderText('Min(0)');
    const maxInput = getByPlaceholderText('Max');
    fireEvent.change(minInput, { target: { value: 'abc' } }); // Invalid input
    fireEvent.change(maxInput, { target: { value: 'test' } }); // Invalid input
    expect(mockProps.onChange).toHaveBeenCalled(); // Expect no call
  });

  it('should handle onChange event for min input (empty value)', () => {
    const { getByPlaceholderText } = render(<RangeTextInput {...mockProps} value={{ "none_range": false, "value": "" }} />);
    const maxInput = getByPlaceholderText('Max');

    fireEvent.change(maxInput, { target: { value: '0' } });

    expect(mockProps.onChange).toHaveBeenCalledWith('num_computers__range', { "none_range": false, "value": "" });
  });


  it('should handle onChange event for max input (valid value)', () => {
    const { getByPlaceholderText } = render(<RangeTextInput {...mockProps} />);
    const maxInput = getByPlaceholderText('Max');

    fireEvent.change(maxInput, { target: { value: '25' } });

    expect(mockProps.onChange).toHaveBeenCalledWith('num_computers__range', { "none_range": true, "value": "15,25" });
  });

  it('should handle onChange event on Show null values', () => {
    mockProps.onChange.mockReset();
    const { getByLabelText } = render(<RangeTextInput {...mockProps} />);
    const checkbox = getByLabelText('Show null values');

    fireEvent.click(checkbox);
    expect(mockProps.onChange).toHaveBeenCalled();
  });

  it('should handle onChange event on Show null value with empty min max', () => {
    mockProps.onChange.mockReset();
    const { getByLabelText } = render(<RangeTextInput {...mockProps} value={{ "none_range": true, "value": "" }} />);
    const checkbox = getByLabelText('Show null values');

    fireEvent.click(checkbox);
    expect(mockProps.onChange).toHaveBeenCalled();
  });

});

// describe('FilterPopup test', () => {
//   it('should render valid mode', () => {
//     const { asFragment } = render(
//       testWrapper(<FilterPopup open={false} setOpen={jest.fn()} />)
//     );
//     expect(asFragment()).toMatchSnapshot();
//   })
//   it('should render valid mode', () => {
//     const { asFragment } = render(
//       testWrapper(<FilterPopup open={true} setOpen={jest.fn()} />)
//     );
//     expect(asFragment()).toMatchSnapshot();
//   })
// })
