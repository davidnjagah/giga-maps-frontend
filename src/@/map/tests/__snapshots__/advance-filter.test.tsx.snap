// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdvancedFilter should render Advance filter 1`] = `
<DocumentFragment>
  .c0 {
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 1rem;
}

.c0 >h3 {
  font-family: Open Sans;
  font-size: 1rem;
  font-style: normal;
  font-weight: 500;
  line-height: 1.5rem;
}

.c0 button svg {
  fill: !important;
}

.c7 {
  width: 100%;
}

.c7 button {
  width: 50%;
  padding-right: 0;
}

.c2 {
  padding: 0.5rem 1rem;
}

.c2 .cds--list-box__field {
  border-bottom: 1px solid;
}

.c2 .cds--list-box__menu-item--active .cds--list-box__menu-item__option {
  color: #222;
}

.c1 {
  max-height: calc(100vh - 20.5rem);
  padding-bottom: 2rem;
}

.c3 {
  padding: 0.5rem 1rem;
}

.c3 .group-input {
  display: flex;
  justify-content: space-between;
  gap: 1rem;
  padding: 0;
}

.c3 .cds--label {
  margin-bottom: 0.5rem;
}

.c6 {
  padding: 0.5rem 1rem;
}

.c6 .cds--list-box__field--wrapper {
  border-bottom: 1px solid var(--cds-border-strong);
}

.c6 .cds--list-box--expanded.cds--list-box--lg .cds--list-box__menu {
  max-block-size: 20rem;
}

.c6 .cds--list-box__menu-item--active .cds--list-box__menu-item__option {
  color: #222;
}

.c6 .cds--checkbox-label-text {
  font-size: 0.75rem;
}

.c4 {
  padding: 0.5rem 1rem;
}

.c5 {
  margin: 0.5rem 0;
}

.c5 .cds--checkbox-label-text {
  font-size: 0.75rem;
}

@media (min-width:769px) {
  .c0 {
    min-width: 23rem;
  }
}

@media (max-width:768px) {
  .c7 {
    position: fixed;
    bottom: 0;
    left: 0;
  }
}

@media (max-width:768px) {
  .c1 {
    max-height: calc(100vh - 13rem);
  }
}

<span
    class="cds--popover"
  >
    <span
      class="cds--popover-content filter-popover-content"
    >
      <div
        class="c0"
      >
        <h3>
          Filter by
        </h3>
        <span
          class="cds--popover-container cds--popover--caret cds--popover--high-contrast cds--popover--top cds--tooltip cds--icon-tooltip"
        >
          <div
            class="cds--tooltip-trigger__wrapper"
          >
            <button
              aria-labelledby="tooltip-:r0:"
              class="cds--btn--icon-only cds--btn cds--btn--md cds--layout--size-md cds--btn--ghost"
              data-testid="filter-close"
              title="Close filter"
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                focusable="false"
                height="18"
                preserveAspectRatio="xMidYMid meet"
                viewBox="0 0 32 32"
                width="18"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M17.4141 16L24 9.4141 22.5859 8 16 14.5859 9.4143 8 8 9.4141 14.5859 16 8 22.5859 9.4143 24 16 17.4141 22.5859 24 24 22.5859 17.4141 16z"
                />
              </svg>
            </button>
          </div>
          <span
            aria-hidden="true"
            class="cds--popover"
            id="tooltip-:r0:"
            role="tooltip"
          >
            <span
              class="cds--popover-content cds--tooltip-content"
            >
              Close
            </span>
            <span
              class="cds--popover-caret"
            />
          </span>
        </span>
      </div>
      <form
        aria-label="filter-form"
        class="cds--form"
      >
        <div
          class="scrollbar-container c1 ps"
        >
          <div
            class="cds--dropdown__wrapper cds--list-box__wrapper c2"
          >
            <label
              class="cds--label"
              for="downshift-:r2:-toggle-button"
              id="downshift-:r2:-label"
            >
              School Region
            </label>
            <div
              class="cds--dropdown cds--dropdown--md cds--list-box cds--list-box--md"
              id="dropdown-environment"
            >
              <button
                aria-activedescendant=""
                aria-controls="downshift-:r2:-menu"
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="downshift-:r2:-label"
                class="cds--list-box__field"
                id="downshift-:r2:-toggle-button"
                role="combobox"
                tabindex="0"
                title="All"
                type="button"
              >
                <span
                  class="cds--list-box__label"
                >
                  All
                </span>
                <div
                  class="cds--list-box__menu-icon"
                >
                  <svg
                    aria-label="Open menu"
                    fill="currentColor"
                    focusable="false"
                    height="16"
                    name="chevron--down"
                    preserveAspectRatio="xMidYMid meet"
                    role="img"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
                    />
                    <title>
                      Open menu
                    </title>
                  </svg>
                </div>
              </button>
              <ul
                aria-labelledby="downshift-:r2:-label"
                class="cds--list-box__menu"
                id="downshift-:r2:-menu"
                role="listbox"
              />
            </div>
          </div>
          <div
            class="cds--dropdown__wrapper cds--list-box__wrapper c2"
          >
            <label
              class="cds--label"
              for="downshift-:r3:-toggle-button"
              id="downshift-:r3:-label"
            >
              School Type
            </label>
            <div
              class="cds--dropdown cds--dropdown--md cds--list-box cds--list-box--md"
              id="dropdown-school_type"
            >
              <button
                aria-activedescendant=""
                aria-controls="downshift-:r3:-menu"
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="downshift-:r3:-label"
                class="cds--list-box__field"
                id="downshift-:r3:-toggle-button"
                role="combobox"
                tabindex="0"
                title="All"
                type="button"
              >
                <span
                  class="cds--list-box__label"
                >
                  All
                </span>
                <div
                  class="cds--list-box__menu-icon"
                >
                  <svg
                    aria-label="Open menu"
                    fill="currentColor"
                    focusable="false"
                    height="16"
                    name="chevron--down"
                    preserveAspectRatio="xMidYMid meet"
                    role="img"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
                    />
                    <title>
                      Open menu
                    </title>
                  </svg>
                </div>
              </button>
              <ul
                aria-labelledby="downshift-:r3:-label"
                class="cds--list-box__menu"
                id="downshift-:r3:-menu"
                role="listbox"
              />
            </div>
          </div>
          <div
            class="cds--dropdown__wrapper cds--list-box__wrapper c2"
          >
            <label
              class="cds--label"
              for="downshift-:r4:-toggle-button"
              id="downshift-:r4:-label"
            >
              Education Level
            </label>
            <div
              class="cds--dropdown cds--dropdown--md cds--list-box cds--list-box--md"
              id="dropdown-education_level"
            >
              <button
                aria-activedescendant=""
                aria-controls="downshift-:r4:-menu"
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="downshift-:r4:-label"
                class="cds--list-box__field"
                id="downshift-:r4:-toggle-button"
                role="combobox"
                tabindex="0"
                title="All"
                type="button"
              >
                <span
                  class="cds--list-box__label"
                >
                  All
                </span>
                <div
                  class="cds--list-box__menu-icon"
                >
                  <svg
                    aria-label="Open menu"
                    fill="currentColor"
                    focusable="false"
                    height="16"
                    name="chevron--down"
                    preserveAspectRatio="xMidYMid meet"
                    role="img"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
                    />
                    <title>
                      Open menu
                    </title>
                  </svg>
                </div>
              </button>
              <ul
                aria-labelledby="downshift-:r4:-label"
                class="cds--list-box__menu"
                id="downshift-:r4:-menu"
                role="listbox"
              />
            </div>
          </div>
          <div
            class="c3"
          >
            <label
              class="cds--label cds--label--no-margin"
            >
              No of Students
            </label>
            <div
              class="c4 group-input"
            >
              <div
                class="cds--form-item cds--text-input-wrapper"
              >
                <div
                  class="cds--text-input__label-wrapper"
                />
                <div
                  class="cds--text-input__field-outer-wrapper"
                >
                  <div
                    class="cds--text-input__field-wrapper"
                  >
                    <input
                      class="cds--text-input cds--text-input--sm cds--layout--size-sm"
                      id="num_students-min-input"
                      placeholder="Min (0)"
                      title="Min (0)"
                      type="number"
                      value=""
                    />
                    <span
                      class="cds--text-input__counter-alert"
                      role="alert"
                    />
                  </div>
                </div>
              </div>
              <div
                class="cds--form-item cds--text-input-wrapper"
              >
                <div
                  class="cds--text-input__label-wrapper"
                />
                <div
                  class="cds--text-input__field-outer-wrapper"
                >
                  <div
                    class="cds--text-input__field-wrapper"
                  >
                    <input
                      class="cds--text-input cds--text-input--sm cds--layout--size-sm"
                      id="num_students-max-input"
                      placeholder="Max (1 Lac)"
                      title="Max (1 Lac)"
                      type="number"
                      value=""
                    />
                    <span
                      class="cds--text-input__counter-alert"
                      role="alert"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="cds--form-item cds--checkbox-wrapper c5"
            >
              <input
                class="cds--checkbox"
                id="checkbox-num_students__range"
                style="padding: 0rem 1rem;"
                type="checkbox"
              />
              <label
                class="cds--checkbox-label"
                for="checkbox-num_students__range"
                title=""
              >
                <span
                  class="cds--checkbox-label-text"
                >
                  Show null values
                </span>
              </label>
              <div
                class="cds--checkbox__validation-msg"
              />
            </div>
          </div>
          <div
            class="c3"
          >
            <label
              class="cds--label cds--label--no-margin"
            >
              Download Speed
            </label>
            <div
              class="c4 group-input"
            >
              <div
                class="cds--form-item cds--text-input-wrapper"
              >
                <div
                  class="cds--text-input__label-wrapper"
                />
                <div
                  class="cds--text-input__field-outer-wrapper"
                >
                  <div
                    class="cds--text-input__field-wrapper"
                  >
                    <input
                      class="cds--text-input cds--text-input--sm cds--layout--size-sm"
                      id="connectivity_speed-min-input"
                      placeholder="Min (MBs)"
                      title="Min (MBs)"
                      type="number"
                      value=""
                    />
                    <span
                      class="cds--text-input__counter-alert"
                      role="alert"
                    />
                  </div>
                </div>
              </div>
              <div
                class="cds--form-item cds--text-input-wrapper"
              >
                <div
                  class="cds--text-input__label-wrapper"
                />
                <div
                  class="cds--text-input__field-outer-wrapper"
                >
                  <div
                    class="cds--text-input__field-wrapper"
                  >
                    <input
                      class="cds--text-input cds--text-input--sm cds--layout--size-sm"
                      id="connectivity_speed-max-input"
                      placeholder="Max (MBs)"
                      title="Max (MBs)"
                      type="number"
                      value=""
                    />
                    <span
                      class="cds--text-input__counter-alert"
                      role="alert"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="cds--form-item cds--checkbox-wrapper c5"
            >
              <input
                class="cds--checkbox"
                id="checkbox-connectivity_speed__range"
                style="padding: 0rem 1rem;"
                type="checkbox"
              />
              <label
                class="cds--checkbox-label"
                for="checkbox-connectivity_speed__range"
                title=""
              >
                <span
                  class="cds--checkbox-label-text"
                >
                  Show null values
                </span>
              </label>
              <div
                class="cds--checkbox__validation-msg"
              />
            </div>
          </div>
          <div
            class="cds--dropdown__wrapper cds--list-box__wrapper c2"
          >
            <label
              class="cds--label"
              for="downshift-:r5:-toggle-button"
              id="downshift-:r5:-label"
            >
              Has Computer Lab
            </label>
            <div
              class="cds--dropdown cds--dropdown--md cds--list-box cds--list-box--md"
              id="dropdown-computer_lab"
            >
              <button
                aria-activedescendant=""
                aria-controls="downshift-:r5:-menu"
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="downshift-:r5:-label"
                class="cds--list-box__field"
                id="downshift-:r5:-toggle-button"
                role="combobox"
                tabindex="0"
                title="All"
                type="button"
              >
                <span
                  class="cds--list-box__label"
                >
                  All
                </span>
                <div
                  class="cds--list-box__menu-icon"
                >
                  <svg
                    aria-label="Open menu"
                    fill="currentColor"
                    focusable="false"
                    height="16"
                    name="chevron--down"
                    preserveAspectRatio="xMidYMid meet"
                    role="img"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
                    />
                    <title>
                      Open menu
                    </title>
                  </svg>
                </div>
              </button>
              <ul
                aria-labelledby="downshift-:r5:-label"
                class="cds--list-box__menu"
                id="downshift-:r5:-menu"
                role="listbox"
              />
            </div>
          </div>
          <div
            class="cds--multi-select__wrapper cds--list-box__wrapper c6"
          >
            <label
              class="cds--label"
              for="downshift-:r6:-toggle-button"
              id="downshift-:r6:-label"
            >
              Coverage Type
            </label>
            <div
              class="cds--multi-select cds--list-box cds--list-box--lg"
              id="mutli-select-dropdown-coverage_type"
            >
              <div
                class="cds--list-box__field--wrapper"
              >
                <button
                  aria-activedescendant=""
                  aria-controls="downshift-:r6:-menu"
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  aria-labelledby="downshift-:r6:-label"
                  class="cds--list-box__field"
                  id="downshift-:r6:-toggle-button"
                  role="combobox"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="cds--list-box__label"
                    id="multiselect-field-label-1"
                  >
                    Select Coverage Type1
                  </span>
                  <div
                    class="cds--list-box__menu-icon"
                  >
                    <svg
                      aria-label="Open menu"
                      fill="currentColor"
                      focusable="false"
                      height="16"
                      name="chevron--down"
                      preserveAspectRatio="xMidYMid meet"
                      role="img"
                      viewBox="0 0 16 16"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
                      />
                      <title>
                        Open menu
                      </title>
                    </svg>
                  </div>
                </button>
              </div>
              <ul
                aria-labelledby="downshift-:r6:-label"
                class="cds--list-box__menu"
                id="downshift-:r6:-menu"
                role="listbox"
              />
            </div>
          </div>
          <div
            class="ps__rail-x"
            style="left: 0px; top: 0px;"
          >
            <div
              class="ps__thumb-x"
              style="left: 0px; width: 0px;"
              tabindex="0"
            />
          </div>
          <div
            class="ps__rail-y"
            style="top: 0px; left: 0px;"
          >
            <div
              class="ps__thumb-y"
              style="top: 0px; height: 0px;"
              tabindex="0"
            />
          </div>
        </div>
        <div
          class="c7"
        >
          <button
            class="cds--btn cds--btn--secondary"
            type="reset"
          >
            Reset Filters
          </button>
          <button
            class="cds--btn cds--btn--primary"
            type="submit"
          >
            Apply
          </button>
        </div>
      </form>
    </span>
    <span
      class="cds--popover-caret"
    />
  </span>
</DocumentFragment>
`;

exports[`RangeTextInput Component RangeTextInput input 1`] = `
.c0 {
  padding: 0.5rem 1rem;
}

.c0 .group-input {
  display: flex;
  justify-content: space-between;
  gap: 1rem;
  padding: 0;
}

.c0 .cds--label {
  margin-bottom: 0.5rem;
}

.c2 {
  padding: 0.5rem 1rem;
}

.c3 {
  margin: 0.5rem 0;
}

.c3 .cds--checkbox-label-text {
  font-size: 0.75rem;
}

.c1 button {
  border: none;
  background: inherit;
  cursor: pointer;
}

.c1 .cds--popover-content {
  background: #181818!important;
}

.c1 svg {
  width: 0.75rem;
  height: 0.75rem;
}

<div>
  <div
    class="c0"
  >
    <label
      class="cds--label cds--label--no-margin"
    >
      No of Teachers
      <span
        class="cds--popover-container cds--popover--caret cds--popover--high-contrast cds--popover--bottom cds--tooltip c1"
      >
        <div
          class="cds--tooltip-trigger__wrapper"
        >
          <button
            aria-labelledby="tooltip-:r9:"
            type="button"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              focusable="false"
              height="16"
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8.5 11L8.5 6.5 6.5 6.5 6.5 7.5 7.5 7.5 7.5 11 6 11 6 12 10 12 10 11zM8 3.5c-.4 0-.8.3-.8.8S7.6 5 8 5c.4 0 .8-.3.8-.8S8.4 3.5 8 3.5z"
              />
              <path
                d="M8,15c-3.9,0-7-3.1-7-7s3.1-7,7-7s7,3.1,7,7S11.9,15,8,15z M8,2C4.7,2,2,4.7,2,8s2.7,6,6,6s6-2.7,6-6S11.3,2,8,2z"
              />
            </svg>
          </button>
        </div>
        <span
          aria-hidden="true"
          class="cds--popover"
          id="tooltip-:r9:"
          role="tooltip"
        >
          <span
            class="cds--popover-content cds--tooltip-content"
          >
            test range input
          </span>
          <span
            class="cds--popover-caret"
          />
        </span>
      </span>
    </label>
    <div
      class="c2 group-input"
    >
      <div
        class="cds--form-item cds--text-input-wrapper"
      >
        <div
          class="cds--text-input__label-wrapper"
        />
        <div
          class="cds--text-input__field-outer-wrapper"
        >
          <div
            class="cds--text-input__field-wrapper"
          >
            <input
              class="cds--text-input cds--text-input--sm cds--layout--size-sm"
              id="num_computers-min-input"
              placeholder="Min(0)"
              title="Min(0)"
              type="number"
              value="15"
            />
            <span
              class="cds--text-input__counter-alert"
              role="alert"
            />
          </div>
        </div>
      </div>
      <div
        class="cds--form-item cds--text-input-wrapper"
      >
        <div
          class="cds--text-input__label-wrapper"
        />
        <div
          class="cds--text-input__field-outer-wrapper"
        >
          <div
            class="cds--text-input__field-wrapper"
          >
            <input
              class="cds--text-input cds--text-input--sm cds--layout--size-sm"
              id="num_computers-max-input"
              placeholder="Max"
              title="Max"
              type="number"
              value="20"
            />
            <span
              class="cds--text-input__counter-alert"
              role="alert"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="cds--form-item cds--checkbox-wrapper c3"
    >
      <input
        checked=""
        class="cds--checkbox"
        id="checkbox-num_computers__range"
        style="padding: 0rem 1rem;"
        type="checkbox"
      />
      <label
        class="cds--checkbox-label"
        for="checkbox-num_computers__range"
        title=""
      >
        <span
          class="cds--checkbox-label-text"
        >
          Show null values
        </span>
      </label>
      <div
        class="cds--checkbox__validation-msg"
      />
    </div>
  </div>
</div>
`;
